components/tidb_query/src/rpn_expr/impl_compare.rs

```diff
@@ -849,4 +885,51 @@ mod tests {
             assert_eq!(output, expected);
         }
     }
+
+    #[test]
+    fn test_greatest_and_least() {
+        let int_cases = vec![
+            (vec![None, None], None, None),
+            (vec![Some(1), Some(-1), None], None, None),
+            (
+                vec![Some(-2), Some(-1), Some(1), Some(2)],
+                Some(2),
+                Some(-2),
+            ),
+            (
+                vec![Some(i64::MIN), Some(0), Some(-1), Some(i64::MAX)],
+                Some(i64::MAX),
+                Some(i64::MIN),
+            ),
+            (vec![Some(0), Some(4), Some(8)], Some(8), Some(0)),
+        ];
+
+        type Cases<T> = Vec<(Vec<Option<T>>, Option<T>, Option<T>)>;
+
+        fn do_test<T>(greatest_sig: ScalarFuncSig, least_sig: ScalarFuncSig, cases: Cases<T>)
+        where
+            T: Debug + PartialEq + Evaluable,
+            Option<T>: From<ScalarValue> + Into<ScalarValue>,
+        {
```

Comment:
you can simply remove these lines and write `ScalarFuncSig::GreatestInt` inline directly.

Reply:
finished